osquery/tables/system/windows/userassist.cpp

```diff
@@ -0,0 +1,189 @@
+    gmtime_s(&tm, &last_run_time);
+
+    auto time_str = platformAsctime(&tm);
+    return time_str;
+  }
+}
+
+// Get execution count
+int execution_num(std::string& assist_data) {
+  if (assist_data.length() <= 16) {
+    LOG(WARNING) << "Userassist execution count format is incorrect";
+    return -1;
+  }
+
+  std::string execution_count = assist_data.substr(8, 8);
+
+  // swap endianess
+  std::reverse(execution_count.begin(), execution_count.end());
+
+  for (std::size_t i = 0; i < execution_count.length(); i += 2) {
+    char temp = execution_count[i];
+    execution_count[i] = execution_count[i + 1];
+    execution_count[i + 1] = temp;
+  }
+
+  auto count = tryTo<std::size_t>(execution_count, 16);
+  if (count.isError()) {
+    LOG(WARNING) << "Error getting execution count: " << count.takeError();
+    return -1;
+  }
+  return count.get();
+}
+
+QueryData genUserAssist(QueryContext& context) {
+  QueryData results;
+  QueryData users;
+
+  queryKey("HKEY_USERS", users);
+
+  for (const auto& uKey : users) {
+    auto keyType = uKey.find("type");
+    auto keyPath = uKey.find("path");
+    if (keyType == uKey.end() || keyPath == uKey.end()) {
+      continue;
+    }
+
+    std::string fullPath = keyPath->second + kFullRegPath;
+    QueryData user_assist_results;
+    queryKey(fullPath, user_assist_results);
+
+    for (const auto& rKey : user_assist_results) {
+      auto keyType = rKey.find("type");
+      auto keyPath = rKey.find("path");
+      if (keyType == rKey.end() || keyPath == rKey.end()) {
+        continue;
+      }
+
+      std::string full_path_key = keyPath->second + "\\Count";
+
+      QueryData assist_results;
+      queryKey(full_path_key, assist_results);
+
+      for (const auto& aKey : assist_results) {
+        std::string subkey = aKey.at("path");
+
+        // split reg path by \Count\ to get Key values
+        std::size_t count_key = subkey.find("Count\\");
+        std::string value_key = subkey.substr(count_key);
+        std::string value_key_reg = value_key.substr(6, std::string::npos);
+
+        std::string decoded_value_key = rot_decode(value_key_reg);
+
+        Row r;
+
+        // set UEME_CTLCUACount:ctor and UEME_CTLSESSION values to blank they
+        // are not executables
+        if (decoded_value_key == "UEME_CTLCUACount:ctor" ||
+            decoded_value_key == "UEME_CTLSESSION") {
+          r["path"] = decoded_value_key;
+          r["last_execution_time"] = "";
+          r["count"] = INTEGER("");
+          r["sid"] = uKey.at("name");
+          results.push_back(r);
+        } else {
+          std::string assist_data = aKey.at("data");
+
+          auto count = execution_num(assist_data);
+          auto time_str = last_execute(assist_data);
+
+          r["path"] = decoded_value_key;
+          r["last_execution_time"] = time_str;
+          if (time_str == "") {
+            r["count"] = "";
+          } else {
+            r["count"] = INTEGER(count);
+          }
```

Comment:
It seems to me like you could bring the function call on line 168 to get the `count` value inside of your `else` condition. If the `time_str == ""`, then we should avoid making the call to `execution_num`.

Reply:
moved function